src/libraries/System.Private.CoreLib/src/System/Diagnostics/CodeAnalysis/ExperimentalAttribute.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace System.Diagnostics.CodeAnalysis;
+
+/// <summary>
+///  Indicates that an API is experimental and it may change in the future. This attribute allows call sites to be flagged with
+///  a diagnostic that indicates that an experimental feature is used. Authors can use this attribute to ship preview features in their assemblies.
+/// </summary>
+[AttributeUsage(AttributeTargets.Assembly |
+                AttributeTargets.Module |
+                AttributeTargets.Class |
+                AttributeTargets.Struct |
+                AttributeTargets.Enum |
+                AttributeTargets.Constructor |
+                AttributeTargets.Method |
+                AttributeTargets.Property |
+                AttributeTargets.Field |
+                AttributeTargets.Event |
+                AttributeTargets.Interface |
+                AttributeTargets.Delegate, Inherited = false)]
+public sealed class ExperimentalAttribute : Attribute
+{
+    /// <summary>
+    ///  Initializes a new instance of the <see cref="ExperimentalAttribute"/> class, specifying the ID that the compiler will use
+    ///  when reporting a use of the API the attribute applies to.
+    /// </summary>
+    /// <param name="diagnosticId">The ID that the compiler will use when reporting a use of the API the attribute applies to.</param>
+    /// <exception cref="ArgumentNullException"><paramref name="diagnosticId"/> is <see langword="null"/>.</exception>
+    /// <exception cref="ArgumentException"><paramref name="diagnosticId"/> is empty.</exception>
+    public ExperimentalAttribute(string diagnosticId)
+    {
+        ArgumentException.ThrowIfNullOrEmpty(diagnosticId);
```

Comment:
```suggestion
```
We generally don't want to throw exceptions from attribute ctors. It makes reflection challenging.